app/controllers/declaration/ProcedureCodesController.scala

```diff
@@ -126,15 +126,25 @@ class ProcedureCodesController @Inject()(
   private def updateCache(itemId: String, updatedProcedureCodes: ProcedureCodesData)(
     implicit r: JourneyRequest[AnyContent]
   ): Future[Option[ExportsDeclaration]] = {
+
+    def clearDataForProcedureCode(code: String, itemId: String, model: ExportsDeclaration) = {
+      var result = model
```

Comment:
There are now 2 Proc Codes that imply that we need to clear possible invalid data
- fiscal info when code !- 1042
- package info when Clearance and code is 0019

There may me a better way to do this without a var.

Reply:
Yeah, this method should be rewrited. I'll try to look at this in the IntelliJ.

Reply:
Proposal, but to be honest I don't have any ideas how to make it more readable a lot of conditional logic...

Some improvements, probably can be better to use
 instead of:
`r.declarationType == DeclarationType.CLEARANCE`
use
`request.isType(CLEARANCE)`

And change `exportInventoryCleansingRecordProcedureCodes` to something shorter. Whit this probably it will be slightly more readable.

```
def clearDataForProcedureCode(code: String, itemId: String, model: ExportsDeclaration): ExportsDeclaration = {
      
      val decWithUpdatedFiscalReferences =
        if (!ProcedureCodesData.osrProcedureCodes.contains(code))
          model.updatedItem(itemId, item => item.copy(fiscalInformation = None, additionalFiscalReferencesData = None))
        else model
      
      if (r.declarationType == DeclarationType.CLEARANCE && ProcedureCodesData.exportInventoryCleansingRecordProcedureCodes.contains(code))
        decWithUpdatedFiscalReferences.updatedItem(itemId, item => item.copy(packageInformation = None))
      else decWithUpdatedFiscalReferences
    }
```

Reply:
So basically passing each (potentially) modified model to the next section/test?   I'm not sure that is more readable than a mutable model (i.e. the var).

Reply:
@PauloPortugal Check this